src/main/java/io/cryostat/net/web/http/api/beta/RecordingMetadataLabelsPostHandler.java

```diff
@@ -116,28 +117,27 @@ public boolean requiresAuthentication() {
     }
 
     @Override
-    public IntermediateResponse<Map<String, String>> handle(RequestParameters params)
-            throws Exception {
+    public IntermediateResponse<Metadata> handle(RequestParameters params) throws Exception {
         String recordingName = params.getPathParams().get("recordingName");
 
         try {
-            Map<String, String> labels =
-                    recordingMetadataManager.parseRecordingLabels(params.getBody());
+            Metadata metadata =
+                    new Metadata(recordingMetadataManager.parseRecordingLabels(params.getBody()));
 
             recordingArchiveHelper.getRecordingPath(recordingName).get();
 
-            Map<String, String> updatedLabels =
-                    recordingMetadataManager.setRecordingLabels(recordingName, labels).get();
+            Metadata updatedMetadata =
+                    recordingMetadataManager.setRecordingMetadata(recordingName, metadata).get();
 
             notificationFactory
                     .createBuilder()
                     .metaCategory(RecordingMetadataManager.NOTIFICATION_CATEGORY)
                     .metaType(HttpMimeType.JSON)
-                    .message(Map.of("recordingName", recordingName, "labels", labels))
+                    .message(Map.of("recordingName", recordingName, "metadata", updatedMetadata))
```

Comment:
Could you also update the -web side with the new notification message format?

Reply:
Oh right, thanks.